qa/rpc-tests/sc_cert_base.py

```diff
@@ -203,31 +199,90 @@ def run_test(self):
+        epoch_block_hash = self.nodes[0].getblockhash(sc_creating_height - 1 + ((epoch_number + 1) * EPOCH_LENGTH))
+        eph_wrong = self.nodes[0].getblockhash(sc_creating_height)
+        print "epoch_number = ", epoch_number, ", epoch_block_hash = ", epoch_block_hash
+ 
+        amount_cert_2 = [{"pubkeyhash": pkh_node1, "amount": 0}]
 
-        bal_before = bal_after
+        bal_before_cert_2 = self.nodes[1].getbalance("", 0)
+        mark_logs("Generate new certificate for epoch {}. No bwt is included".format(epoch_number), self.nodes, DEBUG_MODE)
+        try:
+            cert_epoch_1 = self.nodes[0].send_certificate(scid, epoch_number, epoch_block_hash, amount_cert_2, CERT_FEE)
+            assert(len(cert_epoch_1) > 0)
+            mark_logs("Certificate is {}".format(cert_epoch_1), self.nodes, DEBUG_MODE)
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs("Send certificate failed with reason {}".format(errorString), self.nodes, DEBUG_MODE)
+            assert(False)
 
-        mark_logs("Checking that Node1 can spend coins received from bwd transfer", self.nodes, DEBUG_MODE)
-        mark_logs("Node 1 sends {} coins to node2...".format(bwt_amount / 2), self.nodes, DEBUG_MODE)
-        tx = self.nodes[1].sendtoaddress(self.nodes[2].getnewaddress(), bwt_amount / 2)
+        mark_logs("Confirm the certificate for epoch {},".format(epoch_number), self.nodes, DEBUG_MODE)
+        blocks.extend(self.nodes[0].generate(1))
+        self.sync_all()
+
+        bal_after_cert_2 = self.nodes[1].getbalance("", 0)    
+
+        mark_logs("Checking that certificate received from previous epoch is spendable,".format(epoch_number), self.nodes, DEBUG_MODE)
+        retrieved_cert = self.nodes[1].gettransaction(cert_epoch_0)
+        # print "cert for epoch 0, following its maturity:\n", pprint.pprint(retrieved_cert)
+        assert_equal(retrieved_cert['amount'],               amount_cert_1[0]["amount"])  # Certificate amount has matured
+        assert_equal(retrieved_cert['details'][0]['category'], "receive")
+        assert_equal(retrieved_cert['details'][0]['amount'], amount_cert_1[0]["amount"])  # In cert details you can see the actual amount transferred
+
+        mark_logs("Checking Node1 balance is duly updated,".format(epoch_number), self.nodes, DEBUG_MODE)
+        assert_equal(bal_after_cert_2, bal_before_cert_2  + amount_cert_1[0]["amount"])
+
+        Node2_bal_before_cert_expenditure = self.nodes[2].getbalance("", 0)
+        mark_logs("Checking that Node1 can spend coins received from bwd transfer in previous epoch", self.nodes, DEBUG_MODE)
+        mark_logs("Node 1 sends {} coins to node2...".format(amount_cert_1[0]["amount"] / 2), self.nodes, DEBUG_MODE)
+        tx = self.nodes[1].sendtoaddress(self.nodes[2].getnewaddress(), amount_cert_1[0]["amount"] / 2)
         assert(len(tx) > 0)
```

Comment:
The sendtoaddress cmd above could be performed in a try/catch block instead of asserting len

Reply:
agree, done